packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -702,6 +713,53 @@ class AndroidGradleBuilder implements AndroidBuilder {
       color: TerminalColor.green,
     );
   }
+
+  @override
+  Future<List<String>> getBuildVariants({required FlutterProject project}) async {
+    final Status status = _logger.startProgress(
+      "Running Gradle task 'printBuildVariants'...",
+    );
+    final List<String> command = <String>[
+      _gradleUtils.getExecutable(project),
+      '-q', // suppresses gradle output.
+      'printBuildVariants',
+    ];
+
+    final Stopwatch sw = Stopwatch()
+      ..start();
+    RunResult result;
+    try {
+      result = await _processUtils.run(
+        command,
+        workingDirectory: project.android.hostAppGradleRoot.path,
+        allowReentrantFlutter: true,
+        environment: <String, String>{
+          if (globals.androidSdk?.javaHome != null)
+            'JAVA_HOME': globals.androidSdk!.javaHome!,
```

Comment:
constant here.
If the `if` statement ensures that javaHome is not null then why are the bangs needed? Should you save globals.androidSdk.javaHome to a variable?

Reply:
I assume if we don't use this it would use the one in the system paths. Having this would prioritize the. AndroidStudio one?